### PR TITLE
Cache Dropdowns

### PR DESCRIPTION
**Pull request by sam-warren**
# Overview

This PR includes the following proposed change(s):

- Moved dropdown caching to useDataAccess.tsx, cache dropdowns on login since they are user-specific
- Fixed upsert where ID was not a primary key, drop code_tables table on login since different users see different tables.
- Tested blue dot tracking on mobile - working
- Tested browser cache clear - non-issue
- Updated makefile to make building and running ios simpler (`make ios`)

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

Testing on mobile

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have made corresponding changes to the documentation
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/src/contexts/DatabaseContext.tsx [hunk 1]
```diff
@@ -143,10 +143,11 @@ export const DatabaseContextProvider = (props) => {
     for (const value of enumKeys(DocType)) {
       switch (value) {
         case 'CODE_TABLE':
+          setupSQL += `DROP TABLE IF EXISTS ${DocType[value]};`; // To wipe old table and get latest codes
```
Comment (micheal-w-wells) [inline app/src/contexts/DatabaseContext.tsx]: I see what you're going for but I think what might happen is the code tables getting wiped on startup while out of network.

**Review by micheal-w-wells — COMMENTED** [review]
Looks good but that one gotcha I think we want to address before we merge.

File: app/src/contexts/DatabaseContext.tsx (hunk 1 above)
Comment (sam-warren) [reply app/src/contexts/DatabaseContext.tsx]: Removed

**Pull request merged**